### ISSUE TITLE
Point cloud is not with respect to the center of camera

### ISSUE THREAD

**Issue by ipa-hsd** [ensenso/ros_driver#9]
My setup includes ensenso N35 mounted on a robot's end-effector. When I visualize the point cloud wrt world_frame (which also happens to be the robot's base_link), it appears slightly shifted (around 5cm). It seems like the point cloud is actually wrt one of the lens (near the Ethernet connector, which is also 5cm from the center) and not the center of camera. 

Also would really appreciate if we have a ensenso_description as mentioned in https://github.com/ensenso/ros_driver/issues/3 since we had to manually specify other frames. As reference : [kinect](https://github.com/turtlebot/turtlebot/blob/hydro/turtlebot_description/urdf/sensors/kinect.urdf.xacro)

**Comment by saierd** [comment]
That's correct. By default the point cloud is relative to the (rectified) projection center of the left camera, not to the center of the housing.

Providing this link in a URDF file is not really possible, because we don't actually know the exact position of the projection center to the camera housing (which depends on the camera model and its calibration) and it would be quite difficult to calibrate it precisely (because you would need to mount the camera very precisely to a known position). Besides, even if you knew the position of the mounting point, it would probably be impossible to mount the camera exactly to a certain position.

The correct way to account for this offset, is to perform a [workspace calibration](http://wiki.ros.org/ensenso_driver/Tutorials/WorkspaceCalibration) or a [hand-eye calibration](http://wiki.ros.org/ensenso_driver/Tutorials/HandEyeCalibration).
